File: src/server/gkill/api/req_res/browse_zip_contents_response.go
package req_res

import "github.com/mt3hr/gkill/src/server/gkill/api/message"

type ZipEntry struct {
	Path string `json:"path"`

	IsDir bool `json:"is_dir"`

	Size int64 `json:"size"`

	IsImage bool `json:"is_image"`

	FileURL string `json:"file_url"`
}

type BrowseZipContentsResponse struct {
	Messages []*message.GkillMessage `json:"messages"`

	Errors []*message.GkillError `json:"errors"`

	Entries []*ZipEntry `json:"entries"`
}

File: src/client/classes/api/req_res/browse-zip-contents-response.ts
'use strict'

import { GkillAPIResponse } from '../gkill-api-response'

export interface ZipEntry {
    path: string
    is_dir: boolean
    size: number
    is_image: boolean
    is_text: boolean
    is_video: boolean
    is_audio: boolean
    is_pdf: boolean
    file_url: string
}

export class BrowseZipContentsResponse extends GkillAPIResponse {

    entries: ZipEntry[]

    constructor() {
        super()
        this.entries = []
    }

}
